src/core/lib/security/transport/security_handshaker.cc

```diff
@@ -317,6 +317,9 @@ grpc_error* SecurityHandshaker::OnHandshakeNextDoneLocked(
 void SecurityHandshaker::OnHandshakeNextDoneGrpcWrapper(
     tsi_result result, void* user_data, const unsigned char* bytes_to_send,
     size_t bytes_to_send_size, tsi_handshaker_result* handshaker_result) {
+  // This callback may be invoked by TSI in a non-grpc thread, so it's safe to
+  // create our own exec_ctx here.
+  grpc_core::ExecCtx exec_ctx;
```

Comment:
No need for the `grpc_core::` prefix, since we're already in that namespace here.

Reply:
I suspect that adding a new `exec_ctx` will break ALTS TSI implementation. The `exec_ctx` is removed in  #16773 and the main reason is that since ALTS TSI is now a part of gRPC core which already has its own active `exec_ctx` and creating another will violate the policy - "Only allow one "active" ExecCtx on a thread at the same time".

Reply:
I don't think this will be an issue.  It turns out that we (unfortunately) have the ability to "stack" multiple ExecCtx instances in the same C stack.  We should still avoid this whenever possible, but we can take advantage of it here if there's a benefit to doing so.

It's also worth noting that this entire issue will go away in the not-too-distant future, since @yashykt will be working on a new closure API design that completely eliminates the use of ExecCtx.

Reply:
Thanks Mark for the clarification. I think that also explains why none of ALTS interop tests fail in this PR. When working on #16773, I clearly remembered all interop tests failed if creating a new `exec_ctx` and I believe we do not allow to "stack" multiple ExecCtx instances at the time.

Reply:
I suspect that our ability to "stack" ExecCtx instances predates that PR, but @yashykt can probably tell you for sure.

Reply:
@vjpai Do we need to add ApplicationCallbackExecCtx here? Any suggestion?